sdk/smx.ts

```diff
@@ -143,20 +143,32 @@ export class SMXStage {
    * SET_SERIAL_NUMBERS
    */
 
-  writeConfig(): Promise<AckPacket> {
-    if (!this.info) throw new ReferenceError("this.info does not exist for stage. Cannot write config");
-    if (!this.config) throw new ReferenceError("this.config does not exist for stage. Cannot write empty config");
+  private async needsInfo(): Promise<SMXDeviceInfo> {
+    if (!this.info) await this.updateDeviceInfo();
+    if (!this.info) throw new ReferenceError("this.info does not exist for stage.");
 
-    const command = this.info.firmware_version < 5 ? API_COMMAND.WRITE_CONFIG : API_COMMAND.WRITE_CONFIG_V5;
-    const encoded_config = this.config.encode();
+    return this.info;
+  }
+
+  private async needsConfig(): Promise<SMXConfig> {
+    if (!this.config) await this.updateConfig();
+    if (!this.config) throw new ReferenceError("this.config does not exist for stage.");
+
+    return this.config;
+  }
+
+  async writeConfig(): Promise<AckPacket> {
+    const info = await this.needsInfo();
+    const config = await this.needsConfig();
+
+    const command = info.firmware_version < 5 ? API_COMMAND.WRITE_CONFIG : API_COMMAND.WRITE_CONFIG_V5;
+    const encoded_config = config.encode();
     this.events.output$.push([command, encoded_config.length].concat(encoded_config));
 
     return this.events.ackReports$.firstToPromise();
   }
 
-  setLightStrip(color: RGB): Promise<AckPacket> {
-    if (!this.info) throw new ReferenceError("this.info does not exist for stage. Can not set light strip");
-
+  async setLightStrip(color: RGB): Promise<AckPacket> {
```

Comment:
technically don't need these async keywords when you never use `await` in the function body, but doesn't really matter either way

Reply:
oh right yeah, those might just be a hold out when I was testing a different method earlier.